spec/spec.json

```diff
@@ -50,6 +50,19 @@
             "required": false,
             "url": "https://www.elastic.co/guide/en/ecs/current/ecs-tracing.html",
             "comment": "When APM agents add this field to the context, ecs loggers should pick it up and add it to the log event."
+        },
+        "service.name": {
+            "type": "string",
+            "required": false,
+            "url": "https://www.elastic.co/guide/en/ecs/current/ecs-service.html",
+            "comment": "Configurable by users. When an APM agent is active, they should auto-configure it if not already set."
+        },
+        "event.dataset": {
+            "type": "string",
+            "required": false,
+            "url": "https://www.elastic.co/guide/en/ecs/current/ecs-event.html",
+            "default": "${service.name}.log/${service.name}.${appender.name}",
+            "comment": "Configurable by users. By default derived from the service.name. When logging is auto-configured by APM Agents, they should set this to the log appender name if available. The field helps to filter for different log streams from the same pod, for example and is required for log anomaly detection."
```

Comment:
> When logging is auto-configured by APM Agents,

Is this something all APM agents do? 

> they should set this to the log appender name if available.

The default already includes this? Also who sets this default?

Reply:
> Is this something all APM agents do?

Not as of yet. I guess it makes sense to standardize on that behavior with as part of this issue: https://github.com/elastic/apm/issues/373

> The default already includes this? Also who sets this default?

If the user manually configures the service name, the logging library should set `event.dataset=${service.name}.log` if not configured otherwise.

When agents auto-configure the app to use an ECS logger (https://github.com/elastic/apm/issues/373) they should set `event.dataset=${service.name}.${appender.name}` if there's a method similar to [`Appender#getName`](https://logging.apache.org/log4j/2.x/log4j-core/apidocs/org/apache/logging/log4j/core/Appender.html#getName--). Otherwise, they should also set `event.dataset=${service.name}.log`

Reply:
Cool cool, I am not sure this should be encoded as default or as a documented best practice. 

Only zooming in on this because these are best practices no matter how the logging integration is constructed (e.g APM instrumentation/helper methods or fully manual).

Changing the `/` to `OR`, reads better IMO.

```
"${service.name}.${appender.name}.log OR ${service.name}.log",
```

Reply:
I think even when configuring the logger manually it makes sense make the logger default to `${service.name}.log` so that there are fewer things the user has to configure. But they can obviously override the value if they don't like the default.
I think I prefer `${service.name}.${appender.name}` over `${service.name}.${appender.name}.log` but I like your suggestion to change `/` to `OR`.